### PR TITLE
Add definition

### PR DESCRIPTION
**Pull request by TheWudu**
Add Goltergaul's definition gem and make the SportSession as a first step a Defintion::ValueObject.

This leads to validations about types and so. Also reversed the SPORT_TYPE_MAPPINGS as multiple strings can be the same id (e.g. swimming from runtastic + open_water_swimming from garmin, are both mapped to 18 now.)

### CODE REVIEW

File: lib/models/sport_session.rb [hunk 1]
```diff
@@ -1,19 +1,35 @@
...
+        required 'lat', Definition.Type(Float)
+        required 'lng', Definition.Type(Float)
+        optional 'ele', Definition.Type(Float)
+      end))
     end
 
+    definition(Definition.Keys do
+      required :id, Definition.Type(String)
+      optional :sport_type_id, Definition.Type(Integer)
+      optional :sport_type, Definition.Type(String)
```
Comment (ghsmaniotto) [inline lib/models/sport_session.rb:17]: I prefer having the sport_type as a hash/value object with `id` and `name` fields.

**Review by ghsmaniotto — APPROVED** [review]
Nice to have Definitions here :)

File: lib/models/sport_session.rb (hunk 1 above)
Comment (TheWudu) [reply lib/models/sport_session.rb:17]: I am not sure if we should not just remove the sport_type_id .. is it necessary ? 

I think it makes more sense to get rid of the id entirely, as we do not have a relation-table or similar anyways. So just storing the "name" as "sport_type" should be sufficient. Would also allow to get rid of all the id <-> name conversions .. and the id is not shown anyways anywhere. 

What do you think ?

Comment (ghsmaniotto) [reply lib/models/sport_session.rb:17]: The sport_type_id is used to fetch the activities when searching by sport_type. But as far as I remembered it's not necessary to keep it in the model. The "name" field should be enough.

Comment (TheWudu) [reply lib/models/sport_session.rb:17]: Yes but all the sport-type-id usages can easily be changed to just use "sport_type", no need for an integer id imo.

Suggestion: keep it like this right here now, and i prepare another PR based on this (using definition already) without the sport-type-id so we see if this works out properly or not :-)

**Pull request merged**